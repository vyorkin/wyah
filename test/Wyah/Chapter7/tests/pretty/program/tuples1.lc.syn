Program [Decl "pair" (ELam (Var "a") (ELam (Var "b") (ELam (Var "f") (EApp (EApp (EVar (Var "f")) (EVar (Var "a"))) (EVar (Var "b")))))),Decl "snd" (ELam (Var "p") (EApp (EVar (Var "p")) (EVar (Var "F")))),Decl "fst" (ELam (Var "p") (EApp (EVar (Var "p")) (EVar (Var "T"))))]
