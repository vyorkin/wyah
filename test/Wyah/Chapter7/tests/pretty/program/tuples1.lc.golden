let pair = \a -> \b -> \f -> (f a) b;
let snd = \p -> p F;
let fst = \p -> p T;